### PR TITLE
add recovery when fiat donation fails

### PR DESCRIPTION
**Pull request by rodrigopavezi**

### CODE REVIEW

File: src/components/Donations/DonationShelf/index.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { WagmiConfig } from 'wagmi'
 import { DonationContext, useProcessDonation, ProcessDonationProvider, UserContext } from '@lib'
 import { getWagmiClient } from '../../../lib/wagmiClient'
 import { BreadcrumbStep } from '../../Breadcrumbs'
+import { RecoverDonationProvider } from 'src/lib/recoverDonation'
```
Comment (EricWVGG) [inline src/components/Donations/DonationShelf/index.tsx:7]: This is going to break testing a storybook. If you _must_ import from a relative path, you've got to backtrack to `../../../lib` (but ideally you should just import from `@lib`)

File: src/forms/paymentFiat.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const schema: JsonSchema = {
     },
     month: {
       type: 'string',
-      enum: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '11', '12'],
+      enum: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
```
Comment (EricWVGG) [inline src/forms/paymentFiat.ts:24]: oh my god, was this my fuckup??

**Review by EricWVGG — CHANGES_REQUESTED** [review]
now that I see how small this provider is, I really think we ought to merge it with the other Donations provider… but we're sprinting for the deadline now, so let's revisit it post-launch.

File: src/forms/paymentFiat.ts (hunk 2 above)
Comment (rodrigopavezi) [reply src/forms/paymentFiat.ts:24]: Yeah, I was testing the credit card  flow and it required the 10. haha

**Review by EricWVGG — APPROVED** [review]

**Pull request merged**